presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -598,12 +601,22 @@ else if (elementType.getJavaType() == Slice.class) {
 
     private static ObjectWriteFunction arrayWriteFunction(ConnectorSession session, Type elementType, String baseElementJdbcTypeName)
     {
+        String psqlBaseTypeName = getPgArrayBaseType(baseElementJdbcTypeName);
         return ObjectWriteFunction.of(Block.class, (statement, index, block) -> {
-            Array jdbcArray = statement.getConnection().createArrayOf(baseElementJdbcTypeName, getJdbcObjectArray(session, elementType, block));
+            Array jdbcArray = statement.getConnection().createArrayOf(psqlBaseTypeName, getJdbcObjectArray(session, elementType, block));
             statement.setArray(index, jdbcArray);
         });
     }
 
+    private static String getPgArrayBaseType(String baseArrayTypeName)
+    {
+        if (baseArrayTypeName.matches("timestamp\\(\\d\\)")) {
+            // parametric timestamp is not supported in PSQL JDBC driver: https://github.com/pgjdbc/pgjdbc/issues/1864
```

Comment:
"array of parametric timestamp" ?

Reply:
dropped method. logic moved to `TypeUtils.getArrayElementPgTypeName`